tools/RAiDER/aria/prepFromARIA.py

```diff
@@ -0,0 +1,212 @@
+
+    dst_lat = os.path.join(out_dir, 'lat.geo')
+    dst_lon = os.path.join(out_dir, 'lon.geo')
+    da_lat.to_netcdf(dst_lat)
+    da_lon.to_netcdf(dst_lon)
+    logger.debug('Wrote: %s', dst_lat)
+    logger.debug('Wrote: %s', dst_lon)
+    return dst_lat, dst_lon
+
+
+def getHeights(f:str):
+    ds =  xr.open_dataset(f, group='science/grids/imagingGeometry')
+    hgts = ds.heightsMeta.data.tolist()
+    return hgts
+
+
+def makeLOSFile(f:str, filename:str):
+    """ Create line-of-sight file from ARIA azimuth and incidence layers """
+
+    group   = 'science/grids/imagingGeometry'
+    azFile  = os.path.join(f'NETCDF:"{f}":{group}/azimuthAngle')
+    incFile = os.path.join(f'NETCDF:"{f}":{group}/incidenceAngle')
+
+    az, az_prof = rio_open(azFile, returnProj=True)
+    az          = np.stack(az)
+    az[az == 0] = np.nan
+    array_shp   = az.shape[1:]
+
+    heading     = 90 - az
+    heading[np.isnan(heading)] = 0.
+
+    inc = rio_open(incFile)
+    inc = np.stack(inc)
+
+    hgt = np.arange(inc.shape[0])
+    y   = np.arange(inc.shape[1])
+    x   = np.arange(inc.shape[2])
+
+    da_inc  = xr.DataArray(inc, name='incidenceAngle',
+                coords={'hgt': hgt, 'x': x, 'y': y},
+                dims='hgt y x'.split())
+
+    da_head = xr.DataArray(heading, name='heading',
+                coords={'hgt': hgt, 'x': x, 'y': y},
+                dims='hgt y x'.split())
+
+    ds = xr.merge([da_head, da_inc]).assign_attrs(
+                        crs=str(az_prof['crs']), geo_transform=az_prof['transform'])
+
+
+    dst = f'{filename}.nc'
+    ds.to_netcdf(dst)
+    logger.debug('Wrote: %s', dst)
+    return dst
+
+
+## make not get correct S1A vs S1B
+def getOrbitFile(f:str, out_dir):
+    from eof.download import download_eofs
+    orbit_dir = os.path.join(out_dir, 'orbits')
+    os.makedirs(orbit_dir, exist_ok=True)
+    # import s1reader
+    group ='science/radarMetaData/inputSLC'
+    sats  = []
+    for key in 'reference secondary'.split():
+        ds   = xr.open_dataset(f, group=f'{group}/{key}')
+        slcs = ds['L1InputGranules']
+        ## sometimes there are more than one, but the second is blank
+        ## sometimes there are more than one, but the second is same except version
+        ## error out if its not blank;
+        # assert np.count_nonzero(slcs.data) == 1, 'Cannot handle more than 1 SLC/GUNW'
+        # slc = list(filter(None, slcs))[0].item()
+        slc = slcs.data[0] # temporary hack; not going to use this anyway
+        sats.append(slc.split('_')[0])
+        # orbit_path = s1reader.get_orbit_file_from_dir(slc, orbit_dir, auto_download=True)
+
+    dates = parse_dates_GUNW(f)
+    time  = parse_time_GUNW(f)
+    dts   = [datetime.strptime(f'{dt}T{time}', '%Y%m%dT%H:%M:%S') for dt in dates]
+    paths = download_eofs(dts, sats, save_dir=orbit_dir)
+    return paths
+
+
+## only this one opens the product; need to get lats/lons actually
+def get_bbox_GUNW(f:str, buff:float=1e-5):
```

Comment:
@jlmaurer is raider internally also adding a buffer to give at least more grid point outside the user-requested BBOX? If that is the case we dont need to add twice a buffer and the number here could stay small.

Reply:
Raider adds the needed buffer if using ray-tracing, although we need to check whether the buffer is sufficient at high latitudes.